File: frontend/src/types/printer.ts
export interface Printer {
  id: number;
  name: string;
  profile_id: number | null;
  warehouse_id?: number | null;
  connection_type?: string | null;
  description?: string | null;
  provider?: string | null;
  system_printer_name?: string | null;
  profile?: {
    id: number;
    name: string;
    offset_x_mm: number;
    offset_y_mm: number;
    scale: number;
    dpi?: number | null;
  } | null;
}

File: backend/schemas/printer.py
from pydantic import BaseModel


class PrinterProfileNested(BaseModel):
    id: int
    name: str
    offset_x_mm: float
    offset_y_mm: float
    scale: float
    dpi: int | None = None


class PrinterPayload(BaseModel):
    name: str
    profile_id: int | None = None
    warehouse_id: int | None = None
    connection_type: str | None = None
    description: str | None = None
    provider: str | None = None
    system_printer_name: str | None = None


class PrinterResponse(BaseModel):
    id: int
    tenant_id: int
    name: str
    profile_id: int | None = None
    profile: PrinterProfileNested | None = None
    warehouse_id: int | None = None
    connection_type: str | None = None
    description: str | None = None
    provider: str | None = None
    system_printer_name: str | None = None
    created_at: str | None = None
    updated_at: str | None = None

    class Config:
        from_attributes = True
